src/AutoRest.CSharp/Common/Output/Expressions/Statements/UnaryOperatorStatement.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+
+namespace AutoRest.CSharp.Common.Output.Expressions.Statements
+{
+    internal record UnaryOperatorStatement(ValueExpression Value) : DeclarationStatement;
```

Comment:
I suppose we should make this statement taking the same parameter as `UnaryOperatorStatement` just like we did for InvokeStaticMethodStatement and expression.
We could add an implicit or explicit operator to cast from the expression to statement
Or we should rename this type to something like `ValueExpressionStatement` to represent the cases that we are using a value expression as a statement, since this class now is not take a `UnaryOperatorExpression` therefore it is very likely that this statement will not write as a unary operator thing, it depends on what we passed in.
If we do this change, it is also very reasonable that we could introduce an implicit or explicit operator for this from `ValueExpression` because there is a lot of expression could be standalone statements

Reply:
> I suppose we should make this statement taking the same parameter as UnaryOperatorStatement just like we did for InvokeStaticMethodStatement and expression.

Agreed. Only unary operators that change the variable can be statements, so I would prefer to have `UnaryOperatorStatement(string Operator, ValueExpression Operand)`